data-prepper-plugins/common/src/main/java/org/opensearch/dataprepper/plugins/file/DynamicFileReader.java

```diff
@@ -0,0 +1,12 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.file;
+
+import java.io.InputStream;
+
+public interface DynamicFileReader {
```

Comment:
I tend to think this should be named `FileReader`. We can have an implementation named `DynamicFileReader` which has the logic for choosing the file and then delegating to the known `FileReader`.

We might want to add a `supports(String)` method to determine which type of file each supports.

```
public class DynamicFileReader implements FileReader {
  InputStream getFile(String file) {
    for(FileReader fileReader : allReaders) {
      if(fileReader.supports(file)) {
        return fileReader.getFile(file);
      }
    }
  }
}
```

Reply:
Alternatively, we can move this all into the `opensearch` project for now and think about how to generalize it for `common` later. But, once we make an interface in `common`, we need to take care about breaking the interface until the next major version.